src/reducers/index.js

```diff
@@ -2,11 +2,20 @@ import { combineReducers } from 'redux';
 import loginReducer from './login';
 import postReducer from './posts';
 import shareReducer from './shareReducer';
+import { LOGOUT } from '../constant';
 
-const rootReducer = combineReducers({
+const appReducer = combineReducers({
   user: loginReducer,
   posts: postReducer,
   share: shareReducer,
 });
 
+const rootReducer = (state, action) => {
+  if (action.type === LOGOUT) {
+    localStorage.clear();
+    return undefined;
```

Comment:
Change `return undefined` to:
```
return {
      user: {},
      posts: {
        posts: [],
        newPosts: [],
        newDataAvailable: false,
      },
      share: {},
    };
```

Reply:
this is not needed, if we set state to **undefined** then it will make it 
`return {
      user: {},
      posts: {
        posts: [],
        newPosts: [],
        newDataAvailable: false,
      },
      share: {},
    };`
like this